packages/ui/src/hooks/use-social-connector.ts

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useCallback } from 'react';
+import { useLocation } from 'react-router-dom';
+
+import { signInWithSocial, signInToSoical } from '@/apis/social';
+import { generateRandomString } from '@/utils';
+
+import useApi from './use-api';
+
+const StorageKeyPrefix = 'social_auth_state';
+const WebPlatformPrefix = 'web';
+const MobilePlatformPrefix = 'mobile';
+
+const isMobileWebview = () => {
+  // TODO: read from native sdk imbeded params
+  return true;
+};
+
+const useSocial = () => {
+  const { result: signInWithSocialResult, run: asyncSignInWithSocial } = useApi(signInWithSocial);
+  const { result: signInToSoicalResult, run: asyncSignInToSoical } = useApi(signInToSoical);
```

Comment:
What's the difference between the two?

Reply:
signInWithSocial requests for the social connector login URL from logto core.
signInToSocial requests the actual login with the given social connector code. Any better naming is welcomed.

Reply:
`invokeSocialSignIn` & `completeSocialSignIn`?

Reply:
@xiaoyijun updated the invokeSocialSignIn naming.